src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -289,4 +289,18 @@ public void testCatchWithExplicitFinalVariable() throws IOException {
 
 		assertTrue(content.contains("catch (final java.lang.Exception e)"));
 	}
+
+	@Test
+	public void testCatchWithUnknownExceptions() {
+		// contract: unknown exception type in multicatch should not cause IndexOutOfBoundsException
+		String inputResource = "./src/test/resources/spoon/test/noclasspath/exceptions/Foo.java";
+		Launcher launcher = new Launcher();
+		launcher.addInputResource(inputResource);
+		launcher.getEnvironment().setNoClasspath(true);
+		CtModel model = launcher.buildModel();
+
+		List<CtCatch> catches = model.getElements(new TypeFilter<CtCatch>(CtCatch.class));
+		catches.get(0).getParameter().getType(); // catch with single UnknownException
+		catches.get(1).getParameter().getType(); // multicatch with UnknownException
```

Comment:
could you please add assert, which checks the returned value too?
assertNull(...)